spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Linear.scala

```diff
@@ -114,8 +101,16 @@ class Linear(
       MemoryData.operationWant(forwardPrimDesc, x)
     }
 
-    ParamsShape.weight = realWei
-    ParamsShape.bias = bis
+    require(weight.size().product == realWei.shape.product,
+      s"${getName} weight shape is not correct.")
+
+    // we should resize the tensor. Because sometimes, weight of Linear will has 4-D, where
+    // the last 2 dims is 1. we should reisze it. It will not allocate a new storage because of
+    // the same size.
+    weight.resize(realWei.shape)
+
+    weight.setMemoryData(realWei)
```

Comment:
I think the resize method may be not nessacary... Maybe do the resize in the setMemoryData method?